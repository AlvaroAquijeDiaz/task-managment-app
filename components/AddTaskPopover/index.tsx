import { NextPage } from 'next';
import * as Popover from '@radix-ui/react-popover';
import { ContentStyled } from '../NotificationDialogMenu/Content.styled';
import { TriggerStyled } from '../NotificationDialogMenu/Trigger.styled';
import { SearchBox } from '../Searchbox';
import { SearchboxStyled } from '../Searchbox/Searchbox.styled';

interface FormInputs {
  title: string;
}
export const AddTaskPopover: NextPage = ({ children }) => {
  return (
    <Popover.Root>
      <TriggerStyled
      // bgColor={'#DA584B'}
      // triggerSize={34}
      // p={4}
      // triggerColor="white"
      // rounded={8}
      // hoverBgColor="#fa7c70"
      >
        <span
          style={{
            backgroundColor: '#DA584B',
            width: '34px',
            height: '42px',
            paddingTop: '10px',
            paddingInline: '4px',
            color: 'white',
            borderRadius: '8px',
          }}
        >
          {children}
        </span>
      </TriggerStyled>
      <ContentStyled>
        <SearchboxStyled h={15} placeholder="Task Title" />
      </ContentStyled>
    </Popover.Root>
  );
};
