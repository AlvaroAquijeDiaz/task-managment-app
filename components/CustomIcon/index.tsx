export const CustomIcon = (props: any) => {
  return (
    <svg xmlns="http://www.w3.org/2000/svg" {...props} fill="currentColor">
      <path
        width={props.width}
        height={props.height}
        d="M11 20a9 9 0 110-18 9 9 0 010 18zm0-2a7 7 0 100-14 7 7 0 000 14zm1-7h3v2h-5V6h2v5zM.747 4.282L4.282.747l1.415 1.414L2.16 5.697.747 4.282zM17.717.747l3.536 3.535-1.414 1.415-3.536-3.536L17.718.747h-.001z"
      />
    </svg>
  )
}
