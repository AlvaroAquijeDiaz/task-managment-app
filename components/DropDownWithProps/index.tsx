import {
  DropdownMenuContent,
  DropdownMenuItem,
  DropdownMenuTrigger,
  Root,
} from '@radix-ui/react-dropdown-menu';
import { NextPage } from 'next';
import styled from 'styled-components';
import { ThemeProperties } from '../../styles/theme.config';
import { ExtendedContentProps } from '../NotificationDialogMenu/Content.styled';

// * Since radix-ui has different accessors for the Dropdowns and Popovers, it's needed to restyle them

export const DropdownMenuStyled = styled(
  DropdownMenuContent
)<ExtendedContentProps>`
  border-radius: ${(props) => (props.rounded ? props.rounded : '8px')};
  background-color: ${({ theme }: { theme: ThemeProperties }) =>
    theme.accentBg};
  color: ${({ theme }: { theme: ThemeProperties }) => theme.mainText};
`;

export const DropdownWithProps: NextPage = () => {
  return (
    <Root>
      <DropdownMenuTrigger>Estimate</DropdownMenuTrigger>
      <DropdownMenuStyled rounded={8} style={{ border: '1px solid red' }}>
        <DropdownMenuItem>
          <h1>DropdownMenuItem</h1>
        </DropdownMenuItem>
      </DropdownMenuStyled>
    </Root>
  );
};
