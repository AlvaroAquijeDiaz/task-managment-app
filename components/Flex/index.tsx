import { NextPage } from 'next';
import styled, { CSSProperties } from 'styled-components';

interface FlexProps {
  alignItems?: 'flex-start' | 'center' | 'flex-end';
  justifyContent?: 'center' | 'space-between' | 'space-around';
  gap?: number;
  marginX?: number;
  marginY?: number;
  mt?: number;
  mb?: number;
  direction?: 'row' | 'column';
}

export const Flex = styled.div<FlexProps>`
  flex-direction: ${(props) => (props.direction ? props.direction : 'row')};
  display: flex;
  align-items: ${(props) => (props.alignItems ? props.alignItems : 'center')};
  justify-content: ${(props) => props.justifyContent};
  gap: ${(props) => props.gap}px;
  margin-block: ${(props) => props.marginY}px;
  margin-inline: ${(props) => props.marginX}px;
  margin-top: ${(props) => props.mt}px;
  margin-bottom: ${(props) => props.mb}px;
`;
