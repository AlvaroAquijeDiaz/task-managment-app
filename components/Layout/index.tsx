import { NextPage } from 'next';
import { SearchBox } from '../Searchbox';
import { Sidebar } from '../SideBar';
import { LayoutStyled } from './Layout.styled';

export const Layout: NextPage = ({ children }) => {
  return (
    <>
      <Sidebar />

      <LayoutStyled>
        <SearchBox />
        {children}
      </LayoutStyled>
    </>
  );
};
