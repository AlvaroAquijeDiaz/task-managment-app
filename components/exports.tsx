export * from './CustomIcon/index';
export * from './Flex/index';
export * from './Layout/Layout.styled';
export * from './Layout/index';
export * from './ModalEditAddTask/Datepicker';
export * from './ModalEditAddTask/ModalClose.styled';
export * from './ModalEditAddTask/ModalContent.styled';
export * from './ModalEditAddTask/ModalTrigger.styled';
export * from './ModalEditAddTask/Option.styled';
export * from './ModalEditAddTask/Select.styled';
export * from './ModalEditAddTask/index';
export * from './NotificationDialogMenu/Content.styled';
export * from './NotificationDialogMenu/Notification.styled';
export * from './NotificationDialogMenu/Trigger.styled';
export * from './NotificationDialogMenu/index';
export * from './Searchbox/IconSidebar.styled';
export * from './Searchbox/Searchbox.styled';
export * from './Searchbox/index';
export * from './Selector/Selector.styled';
export * from './Selector/index';
export * from './SideBar/Sidebar.styled';
export * from './SideBar/index';
export * from './Spinner/index';
export * from './TaskBoard/TasksContainer.styled';
export * from './TaskBoard/index';
export * from './TaskBoard/Tags/index';
export * from './TaskBoard/TaskCard/index';
export * from './TaskBoard/TaskWrapper/index';
export * from './TaskBoard/DueDateCard/index';
export * from './TaskBoard/DueDateCard/DueDateCardStyled';
export * from './TaskNavbar/index';
export * from './UserAvatar/index';
