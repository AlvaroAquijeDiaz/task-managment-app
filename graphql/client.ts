import { ApolloClient, InMemoryCache } from '@apollo/client';

// ! Using process.env throws an unexpected error
export const apolloClient = new ApolloClient({
  uri: 'https://syn-api-prod.herokuapp.com/graphql',
  cache: new InMemoryCache(),
  headers: {
    Authorization:
      'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwb3NpdGlvbklkIjoibmVyZGVyeSIsInByb2plY3RJZCI6IjdkZDc4M2U1LWEzNTItNDI0My05ZjM0LTQ4NDNkODdiYTczYSIsImZ1bGxOYW1lIjoiQWx2YXJvIEFxdWlqZSIsImVtYWlsIjoiYWx2YXJvYXF1aWplQHJhdm4uY28iLCJpYXQiOjE2Mzc3MDY4Nzl9.4g_iTVc67xOSscfdJyMAuD0MEMC_9_JbOpXXSu1S4Rg',
  },
});
