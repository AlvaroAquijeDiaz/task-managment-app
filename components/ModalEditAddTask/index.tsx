import { NextPage } from 'next';
import styled from 'styled-components';
import * as Dialog from '@radix-ui/react-dialog';
import { SearchboxStyled } from '../Searchbox/Searchbox.styled';
import { Flex } from '../Flex';
import { GetAllTasksByStatusDocument, Task } from '../../__generated__/graphql-improved';
import { ModalCloseStyled } from './ModalClose.styled';
import { ModalContentStyled } from './ModalContent.styled';
import { ModalTriggerStyled } from './ModalTrigger.styled';
import {
  PointEstimate,
  Scalars,
  Status,
  TaskTag,
  useCreateTaskMutation,
  useUpdateTaskMutation,
} from '../../__generated__/graphql-improved';
import { useForm } from 'react-hook-form';
import toast from 'react-hot-toast';
import { useState } from 'react';
import * as Dropdown from '@radix-ui/react-dropdown-menu';

export const OverlayStyled = styled(Dialog.Overlay)`
  backdrop-filter: brightness(0.7);
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
`;

export interface CustomModalProps {
  task: Partial<Task>;
  type?: 'create' | 'edit';
}

interface MutationData {
  taskName: Scalars['String'];
  pointEstimate: PointEstimate;
  position: Scalars['Float'];
  status: Status;
  tags: Array<TaskTag>;
  dueDate: Date;
}

export const ModalAddEditTask: NextPage<CustomModalProps> = ({
  children,
  type,
  task,
}) => {
  // TODO - refactor this
  const [updateTask, { data, error }] = useUpdateTaskMutation();
  const [createTask, { data: createData, error: createError }] = useCreateTaskMutation();
  const [show, setShow] = useState(false);
  const { register, setValue, handleSubmit } = useForm<MutationData>();

  const onSubmitHandler = handleSubmit(async (data) => {
    const { taskName: name, pointEstimate, dueDate, status, tags } = data;
    try {
      if (type === 'edit') {
        await updateTask({
          variables: {
            updateTaskInput: {
              id: task.id,
              name,
              dueDate,
              pointEstimate,
              tags,
              status,
            },
          },
        });

        toast.success('Task edited!', {
          style: {
            borderRadius: '20px',
            background: '#333',
            color: 'white',
          },
        });
      } else if (type === 'create') {
        await createTask({
          variables: {
            createTaskInput: {
              dueDate,
              name,
              pointEstimate,
              status: Status.Backlog,
              tags: [TaskTag.Rails, TaskTag.React],
            },
          },
          refetchQueries: [GetAllTasksByStatusDocument],
        });

        toast.success('Task created successfully');
        // not permanent solution
        window.location.reload();
      }
    } catch (error) {
      toast.error('Something went wrong');
    }

    console.log({ data, error, createData, createError });
  });

  return (
    <Dialog.Root>
      <OverlayStyled />
      <ModalTriggerStyled p={4}>{children}</ModalTriggerStyled>

      <ModalContentStyled p={25} rounded={10}>
        <Dialog.Title>
          <SearchboxStyled
            autoFocus={false}
            type={'text'}
            p={2}
            fontWeight="700"
            placeholder={task.name || 'New task name'}
            fontSize={17}
            height={20}
            {...register('taskName', { value: task.name })}
          />
        </Dialog.Title>

        <Flex
          gap={16}
          alignItems="center"
          justifyContent="space-between"
          style={{ position: 'relative' }}
        >
          <Dropdown.Root>
            <Dropdown.Trigger>{children}</Dropdown.Trigger>
            <Dropdown.Content>
              {Object.values(PointEstimate).map((point) => (
                <Dropdown.Item
                  key={point}
                  onClick={() => {
                    setValue('pointEstimate', point);
                  }}
                >
                  {point}
                </Dropdown.Item>
              ))}
            </Dropdown.Content>
          </Dropdown.Root>
          <div>
            <span>Due Date</span>
            <input type="date" {...register('dueDate', { required: true })} />
          </div>
        </Flex>

        <Flex gap={24} alignItems="center">
          <ModalCloseStyled p={8}>Cancel</ModalCloseStyled>
          <ModalCloseStyled
            p={8}
            variant="primary"
            onClick={() => onSubmitHandler()}
            type="submit"
          >
            {type}
          </ModalCloseStyled>
        </Flex>
      </ModalContentStyled>
    </Dialog.Root>
  );
};
