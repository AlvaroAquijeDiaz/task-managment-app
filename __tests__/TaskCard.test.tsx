import TestRenderer from 'react-test-renderer';
import { TaskCard } from '../components/exports';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import faker from 'faker';
import {
  GetAllTasksByStatusDocument,
  PointEstimate,
  Status,
  Task,
  TaskTag,
  UserType,
} from '../__generated__/graphql-remastered';

const mocks = [
  {
    request: {
      query: GetAllTasksByStatusDocument,
      variables: {
        input: {
          status: Status.Backlog,
        },
      },
    },

    result: {
      data: {
        tasks: [
          {
            id: faker.random.uuid(),
            name: 'new task',
            createdAt: faker.date.recent(),
            dueDate: faker.date.future(),
            tags: [TaskTag.ANDROID, TaskTag.IOS],
            creator: {
              id: faker.random.uuid(),
              fullName: faker.name.findName(),
              createdAt: faker.date.recent(),
              email: faker.internet.email(),
              updatedAt: faker.date.recent(),
              avatar: faker.image.avatar(),
              type: UserType.Candidate,
            },
            pointEstimate: PointEstimate.ONE,
          },
        ],
      },
    },
  },
];
it('should render a task card', () => {
  const taskCard = TestRenderer.create(
    <MockedProvider mocks={mocks}>
      <TaskCard task={mocks[0].result.data.tasks[0]} />
    </MockedProvider>,
  );

  const tree = taskCard;
  expect(tree.toJSON()).toMatchSnapshot();
});
