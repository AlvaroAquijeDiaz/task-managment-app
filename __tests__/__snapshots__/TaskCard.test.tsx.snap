// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a task card 1`] = `
<div
  className="Flex-sc-clust5-0 ivJocd"
  direction="column"
  style={
    Object {
      "minWidth": "100%",
    }
  }
>
  <div
    className="Flex-sc-clust5-0 kNIbvG"
  >
    <p
      style={
        Object {
          "fontWeight": "bolder",
        }
      }
    >
      new task
    </p>
  </div>
  <div
    className="Flex-sc-clust5-0 FApbX"
  >
    <p
      style={
        Object {
          "fontWeight": 500,
        }
      }
    >
      1
       pts
    </p>
    <div
      className="DueDateCardStyled-sc-dx4o1y-0 kiBuSk"
    >
      <svg
        fill="currentColor"
        height={20}
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11 20a9 9 0 110-18 9 9 0 010 18zm0-2a7 7 0 100-14 7 7 0 000 14zm1-7h3v2h-5V6h2v5zM.747 4.282L4.282.747l1.415 1.414L2.16 5.697.747 4.282zM17.717.747l3.536 3.535-1.414 1.415-3.536-3.536L17.718.747h-.001z"
          height={20}
          width={20}
        />
      </svg>
      <p
        style={
          Object {
            "textTransform": "uppercase",
          }
        }
      >
        Dec 23 2021
      </p>
    </div>
  </div>
  <div
    className="Flex-sc-clust5-0 hScVHi"
    wrap="wrap"
  >
    <div
      className="Tags__TagStyled-sc-17a5l82-0 YDMhO"
    >
      ANDROID
    </div>
    <div
      className="Tags__TagStyled-sc-17a5l82-0 kmwFFy"
    >
      IOS
    </div>
  </div>
  <div
    className="Flex-sc-clust5-0 FApbX"
  >
    <div
      style={
        Object {
          "borderRadius": "50%",
          "overflow": "hidden",
        }
      }
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "inline-block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "width": "initial",
          }
        }
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        >
          <img
            alt=""
            aria-hidden={true}
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjUiIGhlaWdodD0iMjUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
            style={
              Object {
                "background": "none",
                "border": 0,
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          />
        </span>
        <img
          alt="avatar"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": "cover",
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
        <noscript>
          <img
            alt="avatar"
            data-nimg="intrinsic"
            decoding="async"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Favatars.dicebear.com%2Fapi%2Fbottts%2FTracey%20Connelly.svg&w=64&q=75"
            srcSet="/_next/image?url=https%3A%2F%2Favatars.dicebear.com%2Fapi%2Fbottts%2FTracey%20Connelly.svg&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.dicebear.com%2Fapi%2Fbottts%2FTracey%20Connelly.svg&w=64&q=75 2x"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": "cover",
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </noscript>
      </span>
    </div>
  </div>
</div>
`;
